packages/gcf-utils/src/logging-octokit-plugin.ts

```diff
@@ -0,0 +1,245 @@
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/reviews/:review_id/dismissals': {
+    PUT: GitHubActionType.PR_DISMISS_REVIIEW,
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/merge': {
+    PUT: GitHubActionType.PR_MERGE,
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/update-branch': {
+    PUT: GitHubActionType.PR_UPDATE_BRANCH,
+  },
+};
+
+function parseActionValue(
+  actionType: GitHubActionType,
+  options: {
+    [key: string]: string | number;
+  }
+): string {
+  // optional properties for issues.update
+  const allIssueUpdateProps = [
+    'body',
+    'state',
+    'assignees',
+    'labels',
+    'milestone',
+    'title',
+  ];
+  const hasIssueUpdateProps = allIssueUpdateProps.filter(
+    (prop: string) => options[prop]
+  );
+
+  switch (actionType) {
+    case GitHubActionType.ISSUE_ADD_LABELS:
+      return String(options.labels);
+    case GitHubActionType.ISSUE_CREATE_COMMENT:
+      return String(options.body);
+    case GitHubActionType.ISSUE_REMOVE_LABEL:
+    case GitHubActionType.ISSUE_CREATE_LABEL:
+    case GitHubActionType.ISSUE_DELETE_LABEL:
+      return String(options.name);
+    case GitHubActionType.ISSUE_UPDATE_LABEL:
+      return `${options.current_name} to ${options.name}`;
+    case GitHubActionType.ISSUE_UPDATE:
+      return 'updated: ' + hasIssueUpdateProps.join(',');
+    case GitHubActionType.ISSUE_CREATE:
+      return String(options.title);
+    case GitHubActionType.PR_DISMISS_REVIIEW:
+      return `dismiss ${options.review_id}: ${options.message}`;
+    case GitHubActionType.PR_MERGE:
+    case GitHubActionType.PR_UPDATE_BRANCH:
+    default:
+      return 'NONE';
+  }
+}
+
+/**
+ * Parses the outgoing GitHub request to determine the details of the action being taken
+ * @param options options from outgoing request
+ */
+function parseActionDetails(options: {
+  [key: string]: string | number;
+}): GitHubActionDetails {
+  const actionType: GitHubActionType =
+    ActionEndpoints[options.url][options.method] || GitHubActionType.UNKNOWN;
+
+  const details: GitHubActionDetails = {};
+  details.value = parseActionValue(actionType, options);
+
+  if (options['issue_number']) {
+    details.destObjType = GitHubObjectType.ISSUE;
+    details.destObjId = options['issue_number'];
+  } else if (options['pull_number']) {
+    details.destObjType = GitHubObjectType.PR;
+    details.destObjId = options['pull_number'];
+  }
+
+  details.repoName = String(options['repo']);
+  details.repoOwner = String(options['owner']);
+  details.type = actionType;
+
+  return details;
+}
+
+/**
+ * Log a GitHub action as a GitHubActionLog
+ * @param logger logger instance
+ * @param details action details
+ */
+function logGithubAction(logger: MetricLogger, details: GitHubActionDetails) {
+  if (!details.type || details.type === GitHubActionType.UNKNOWN) {
+    // don't log unknown action types
+    return;
+  }
+  const general = {
+    type: details.type,
+    value: details.value || 'UNKNOWN',
```

Comment:
Minor: it might be easier using a `const` for these in case you have typos in the future